### PR TITLE
CTC loss and Language Model

### PR DESCRIPTION
**Pull request by oak11**
I have made changes to the code in DeepSpeech.ipynb --> deepspeech.py such that it is compatible with tensorflow version 0.9

### CODE REVIEW

File: deepspeech.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    layer_2 = tf.nn.dropout(layer_2, keep_prob)
+    #Hidden layer with clipped RELU activation and dropout
+    layer_3 = tf.minimum(tf.nn.relu(tf.add(tf.matmul(layer_2, _weights['h3']), _biases['b3'])), relu_clip)
+    layer_3 = tf.nn.dropout(layer_3, keep_prob)
+
+    # Define lstm cells with tensorflow
+    # Forward direction cell
+    lstm_fw_cell = tf.nn.rnn_cell.BasicLSTMCell(n_cell_dim, forget_bias=1.0, state_is_tuple=False)
+    # Backward direction cell
+    lstm_bw_cell = tf.nn.rnn_cell.BasicLSTMCell(n_cell_dim, forget_bias=1.0, state_is_tuple=False)
```
Comment (oak11) [inline deepspeech.py]: 1) for line 72 and 74, a warning is raised:    '''WARNING:tensorflow:<tensorflow.python.ops.rnn_cell.BasicLSTMCell object at 0x7fc0b4f15650>: Using a concatenated state is slower and will soon be deprecated.  Use state_is_tuple=True. '''
Should I change it to True and use c_state and m_state ?

File: deepspeech.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    x_ix = tf.constant(x_ix, tf.int64)
+    x_val = tf.constant(x_val, tf.int32)
+    x_shape = tf.constant(x_shape, tf.int64)
+    return tf.SparseTensor(x_ix, x_val, x_shape)
+
+labels = [[0]*n_steps]*batch_size         #read from y and initialize this list?
+labels = SimpleSparseTensorFrom(labels)
+loss = ctc_loss(layer_6, labels, [batch_size]*n_steps)
+print loss
+
```
Comment (oak11) [inline deepspeech.py]: Here, the labels have to be taken from y defined on line 30? or just create a list here with A-Z, space, apostrophe, blank, etc?

File: deepspeech.py (hunk 1 above)
Comment (kdavis-mozilla) [inline deepspeech.py]: I'd say yes. It's amazing how quick this API is changing.

**Comment by kdavis-mozilla** [comment]
Just a general comment. I'd rather we, in the near term, keep the python notebook text as it will allow people to on-board quicker.

File: deepspeech.py (hunk 2 above)
Comment (kdavis-mozilla) [inline deepspeech.py]: The labels should be derived from the target sentences, the y defined on line 30. I think some, likely not all, of how it I thought it should work were in an email I sent around May 31. Maybe that'd be a good place to start, but again I've not tried the ideas in the email of May 31.
